core/domain/rights_manager.py

```diff
@@ -547,359 +546,252 @@ def _get_activity_rights(activity_type, activity_id):
-        return (rights_object.community_owned or
-                self.user_id in rights_object.editor_ids or
-                self.user_id in rights_object.owner_ids)
-
-    def _has_viewing_rights(self, rights_object):
-        return (rights_object.status != ACTIVITY_STATUS_PRIVATE or
-                self.user_id in rights_object.viewer_ids or
-                self.user_id in rights_object.editor_ids or
-                self.user_id in rights_object.owner_ids)
-
-    def _can_play(self, rights_object):
-        if rights_object.status == ACTIVITY_STATUS_PRIVATE:
-            return (self._has_viewing_rights(rights_object)
-                    or rights_object.viewable_if_private
-                    or self.is_moderator())
-        else:
-            return True
+def check_can_access_activity(user_id, user_actions, activity_rights):
+    """Checks whether the user can access given activity.
 
-    def _can_edit(self, rights_object):
-        return (
-            self._has_editing_rights(rights_object) or self.is_admin() or (
-                self.is_moderator() and (
-                    rights_object.status != ACTIVITY_STATUS_PRIVATE)
-            )
-        )
+    Args:
+        user_id: str or None. Id of the given user.
+        user_actions: list(str). List of actions given user can perform.
+        activity_rights: rights_object or None. Rights object of the given
+            activity.
 
-    def _can_delete(self, rights_object):
-        is_deleting_own_private_object = (
-            rights_object.status == ACTIVITY_STATUS_PRIVATE and
-            self._is_owner(rights_object))
+    Returns:
+        bool. Whether the given activity can be accessed.
+    """
+    if activity_rights is None:
+        return False
+    elif activity_rights.is_published():
+        return bool(
+            role_services.ACTION_PLAY_ANY_PUBLIC_ACTIVITY in user_actions)
+    elif activity_rights.is_private():
+        return bool(
+            (role_services.ACTION_PLAY_ANY_PRIVATE_ACTIVITY in user_actions) or
+            activity_rights.is_viewer(user_id) or
+            activity_rights.is_owner(user_id) or
+            activity_rights.is_editor(user_id) or
+            activity_rights.viewable_if_private)
 
-        is_mod_deleting_public_object = (
-            rights_object.status == ACTIVITY_STATUS_PUBLIC and
-            self.is_moderator())
 
-        return (
-            is_deleting_own_private_object or is_mod_deleting_public_object)
+def check_can_edit_activity(user_id, user_actions, activity_rights):
+    """Checks whether the user can edit given activity.
 
-    def is_owner(self, activity_type, activity_id):
-        """Returns whether this user is an owner of the given activity.
+    Args:
+        user_id: str or None. Id of the given user.
+        user_actions: list(str). List of actions the user can perform.
+        activity_rights: rights_object or None. Rights object of the given
```

Comment:
oh.

(These were taken out from the decorators. So, there are tests for decorators for checking these functionalities but no separate tests for these)